tests/test_core.py

```diff
@@ -1,10 +1,95 @@
 from unittest.mock import Mock, patch
 from core import ExactDelayPathfinder
+import unittest
+import networkx as nx
 
-@patch('exactdelaypathfinder.core.ExactDelayPathfinder', autospec=True)
-def test_search(mock_EDPF):
-    new_mock_EDPF = mock_EDPF.return_value
-    new_mock_EDPF.search.return_value = 42
-    G = Mock()
-    new_mock_EDPF.search(G, 10, 'a', 'b')
-    new_mock_EDPF.search.assert_called()
\ No newline at end of file
+class TestEDPF(unittest.TestCase):
+
+    @patch('exactdelaypathfinder.core.ExactDelayPathfinder', autospec=True)
+    def test_search(self, mock_EDPF):
+        """ Tests the search function 
+        
+            Parameters:
+               mock_EDPF: Mock object reference of ExactDelayPathfinder
+        """
+        new_mock_EDPF = mock_EDPF.return_value
+        new_mock_EDPF.search.return_value = 42
+        G = Mock()
+        new_mock_EDPF.search(G, 10, 'a', 'b')
+        new_mock_EDPF.search.assert_called()
+        
+    @patch('exactdelaypathfinder.core.ExactDelayPathfinder', autospec=True)
+    def test_search(self, mock_EDPF):
+        """ Tests the _search function, the DFS algorithm (recursive method) 
+        
+            Parameters:
+               mock_EDPF: Mock object reference of ExactDelayPathfinder
+        """
+        new_mock_EDPF = mock_EDPF.return_value
+        new_mock_EDPF._search.return_value = 86
+        G = Mock()
+        delay = 64;
+        start = 'a'
+        end = 'b'
+        new_mock_EDPF._search(G, 64, 'a', 'b', [])
+        new_mock_EDPF._search.assert_called_with(G, delay, start, end, visits=[])
+
```

Comment:
This is how to mock a method without mocking the entire class:
```suggestion
    @patch.object(ExactDelayPathfinder, '_search')
    def test_search(self, mock_search):
        """ Tests the _search function, the DFS algorithm (recursive method) 
        
            Parameters:
               mock_EDPF: Mock object reference of ExactDelayPathfinder
        """
        new_EDPF = ExactDelayPathfinder()
        mock_search.return_value = []

        G = Mock()
        G.nodes = []
        delay = 64;
        start = 'a'
        end = 'b'
        new_EDPF.search(G, 64, 'a', 'b')
        new_EDPF._search.assert_called()
```